camus-shopify/src/main/scala/com/linkedin/camus/shopify/LateArrivingDataMonitor.scala

```diff
@@ -3,47 +3,86 @@ package com.linkedin.camus.shopify
     head("Camus Late Arriving Data Checker", "")
     note(
-      "This Job checks for late arrivig data in Camus drops.")
+      "This Job checks for late arriving data in Camus drops.")
     help("help") text "Prints this usage text"
 
     opt[String]('c', "camus-properties-file") required() valueName "<path>" action { (x, p) =>
       p.copy(camusPropertiesFilePath = x)
     } text "Camus configuration properties file path."
 
+    opt[String]('t', "check-type") required() action { (x, p) =>
+      p.copy(checkType = x)
+    } text "Camus check type: \"runs\" or \"window\""
+
     opt[Long]('w', "window") valueName "<window>" action { (x, p) =>
       p.copy(window = x)
-    } text "Camus configuration properties file path."
+    } text "Hours before last offset to check: only applicable in \"window\" type check"
+
+    opt[Int]('r', "num-runs") optional() valueName "<int>" action { (x, p) =>
+      p.copy(numRuns = x)
+    } text "Number of camus runs to check: only applicable in \"runs\" type check."
+
   }
 
 
   def checkDropsInWindow(properties: Properties, fs: FileSystem, window: Long): Boolean = {
+    log.info(s"Running check on folders dropped in the window of $window milliseconds.")
     log.info("Fetching topics and folders to check...")
     val camusExecutions = new CamusExecutions(properties, fs)
-    log.info(s"Checking folders in window ($window) to flag violation...")
     val foldersToCheck = camusExecutions.droppedFoldersInWindow(window)
+
+    log.info(s"Checking folders to flag violation...")
+    checkDrops(properties, fs, foldersToCheck.toList)
+  }
+
+  def checkDropsInRuns(properties: Properties, fs: FileSystem, runs: Int): Boolean = {
+    log.info(s"Running check on folders dropped in the last $runs Camus runs.")
+    val camusExecutions = new CamusExecutions(properties, fs)
+    val camusRunPaths = camusExecutions.lastRunDirs(runs)
+
+    log.info(s"Collecting directories touched in $runs last Camus runs: $camusRunPaths")
+    val foldersToCheck = camusExecutions.droppedFoldersInRuns(camusRunPaths)
+
+    log.info(s"Checking folders to flag violation...")
+    checkDrops(properties, fs, foldersToCheck.toList)
+  }
+
+  def checkDrops(properties: Properties, fs: FileSystem, foldersToCheck: List[String]): Boolean = {
     var violationFound = false
     foldersToCheck.foreach(
       folder => {
+        log.info(s"Checking $folder")
         val camusDrop = new CamusHourlyDrop(new Path(folder), fs)
 
         if (camusDrop.hasFlag && camusDrop.isFlagViolated) {
           violationFound = true
           log.error(s"Violation found in ${camusDrop.path}")
+          try {
+            val count = camusDrop.violationCount
+            log.error(s"=> late-arriving records for dir ${camusDrop.topicDir} in ${camusDrop.path}: $count")
+            StatsdReporter.gauge(properties, "late-arriving-data", 1L, s"directory:${camusDrop.topicDir}")
```

Comment:
We should probably report count to Datadog too, and it'd be even more awesome to have `camusDrop.totalRecordCount` or something so we can calculate the percentage of LAD for the folder.

Reply:
I thought about it, just the count is meaningless without total count and I figured it could wait and we could do it in a separate PR.